app/code/Magento/Directory/Setup/Patch/Data/AddDataForAustralia.php

```diff
@@ -45,12 +49,17 @@ public function __construct(
      */
     public function apply()
     {
-        /** @var DataInstaller $dataInstaller */
-        $dataInstaller = $this->dataInstallerFactory->create();
-        $dataInstaller->addCountryRegions(
-            $this->moduleDataSetup->getConnection(),
-            $this->getDataForAustralia()
-        );
+        /** @var \Magento\Directory\Model\ResourceModel\Region\Collection $collection */
+        $collection = $this->regionCollectionFactory->create();
```

Comment:
According to implementation `$this->regionCollectionFactory` is null here and will cause fatal error. Looks like you missed to inject regionCollectionFactory dependency via constructor